src/Components/Web.JS/src/Platform/WebAssemblyResourceLoader.ts

```diff
@@ -165,6 +165,12 @@ async function getCacheToUseIfEnabled(bootConfig: BootJsonData): Promise<Cache |
     return null;
   }
 
+  // cache integrity is compromised if the first request has been served over http
+  // in this case, we want to disable caching and integrity validation
+  if (document.location.protocol === 'http:') {
```

Comment:
Would it be better to require https instead? For e.g.

```suggestion
  if (document.location.protocol !== 'https:') {
```

Reply:
You might be right, specifically requiring https opposed to saying anything that isn't http. I'll change that now.